M2/Macaulay2/m2/expressions.m2

```diff
@@ -1174,16 +1168,17 @@ texMath Table := m -> (
 	"\\end{array}}")
 )
 
+texMathTable := lookup(texMath,Table);
 texMath MatrixExpression := m -> (
     if all(m,r->all(r,i->class i===ZeroExpression)) then "0"
-    else if m#?0 then if #m#0>10 then "{\\left(" | texMath(new Table from toList m) | "\\right)}" -- the extra {} is to discourage line breaks
+    else if m#?0 then if #m#0>10 then "{\\left(" | texMathTable m | "\\right)}" -- the extra {} is to discourage line breaks
      else concatenate(
       	      "\\begin{pmatrix}" | newline,
      	      between(///\\/// | newline, apply(toList m, row -> concatenate between("&",apply(row,texMath)))),
 	      "\\end{pmatrix}" -- notice the absence of final \\ -- so lame. no newline either in case last line is empty
 	      )
 	  )
-texMath MatrixDegreeExpression := x -> texMath MatrixExpression x#0 -- degrees not displayed atm
+texMath MatrixDegreeExpression := (lookup(texMath,MatrixExpression))@@first -- degrees not displayed atm
```

Comment:
It might be better to handle this with inheritance by defining
```
MatrixDegreeExpression = new HeaderType of MatrixExpression
```

Reply:
yeah I don't remember why I made that change 2 years ago.
rethinking and possibly reverting

Reply:
first note that the change, as far as I can tell, is purely cosmetic: the end result is the same.
(I will revert it because I don't remember why I did it).
concerning your remark: I guess the problem is with the `#0` or `@@first`. Ideally you're right we would change the way `MatrixExpression` is internally so `MatrixDegreeExpression` can be a descendant of it. At the moment there's no way to squeeze in the information about the degrees in the way `MatrixExpression` is formatted. Since I didn't want to change `MatrixExpression` but still wanted an expression that preserves degrees (since they are printed in `Standard` mode), I created this distinct type.
Again this is all changes made 2 years ago.
That being said, I'm open to suggestions for an alternative format.